cluster/gce/util.sh

```diff
@@ -3925,41 +4226,98 @@ function kube-down() {
     delete-partitionserver ${server_name}
   done
 
+  if [[ "${SCALEOUT_CLUSTER:-false}" == "true" ]]; then
+    for (( tp_num=1; tp_num<=${SCALEOUT_TP_COUNT}; tp_num++ ))
+    do
+      server_name="${INSTANCE_PREFIX}-tp-${tp_num}-master"
+      export ARKTOS_SCALEOUT_SERVER_TYPE="tp"  
+      export CLUSTER_NAME="${INSTANCE_PREFIX}-tp-${tp_num}"
+      export MASTER_NAME="${CLUSTER_NAME}-master"
+      export NODE_INSTANCE_PREFIX="${CLUSTER_NAME}-minion"
+      export WINDOWS_NODE_INSTANCE_PREFIX="${CLUSTER_NAME}-windows-node"
+      echo "Tear down SCALEOUTSERVER:${server_name}"
+      delete-scaleoutserver ${server_name}
+      NODE_TAG="${INSTANCE_PREFIX}-tp-${tp_num}-minion"
+      delete-firewall-rules \
+      "${NODE_TAG}-http-alt" \
+      "${NODE_TAG}-nodeports"
+      KUBECONFIG="${RESOURCE_DIRECTORY}/kubeconfig${KUBEMARK_PREFIX}.tp-${tp_num}"
+      export CONTEXT="${PROJECT}_${CLUSTER_NAME}"
+      clear-kubeconfig
+      rm -rf "${KUBECONFIG}"
+    done
+
+    for (( rp_num=1; rp_num<=${SCALEOUT_RP_COUNT}; rp_num++ ))
+    do
+      server_name="${INSTANCE_PREFIX}-rp-${rp_num}-master"
+      export ARKTOS_SCALEOUT_SERVER_TYPE="tp"  
+      export CLUSTER_NAME="${INSTANCE_PREFIX}-rp-${rp_num}"
+      export MASTER_NAME="${CLUSTER_NAME}-master"
+      export NODE_INSTANCE_PREFIX="${CLUSTER_NAME}-minion"
+      export WINDOWS_NODE_INSTANCE_PREFIX="${CLUSTER_NAME}-windows-node"
+      echo "Tear down SCALEOUTSERVER:${server_name}"
+      delete-scaleoutserver ${server_name}
+      NODE_TAG="${INSTANCE_PREFIX}-rp-${rp_num}-minion"
+      delete-firewall-rules \
+      "${NODE_TAG}-http-alt" \
+      "${NODE_TAG}-nodeports"
+      KUBECONFIG="${RESOURCE_DIRECTORY}/kubeconfig${KUBEMARK_PREFIX}.rp-${rp_num}"
+      export CONTEXT="${PROJECT}_${CLUSTER_NAME}"
+      clear-kubeconfig
+      rm -rf "${KUBECONFIG}"
+    done
+
+    export SCALEOUT_PROXY_NAME="${KUBE_GCE_INSTANCE_PREFIX}-proxy"
+    
 
-  # Un-register the master replica from etcd and events etcd.
-  remove-replica-from-etcd 2379
-  remove-replica-from-etcd 4002
+    
+    delete-proxy
```

Comment:
haproxy operations should be separated from the kube-up. ideally this should be in 130 release.

Reply:
moved this logic to Kube-down.sh